core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -60,10 +60,7 @@ class StorageLevel private(
   assert(replication < 40, "Replication restricted to be less than 40 for calculating hash codes")
 
   if (useOffHeap) {
-    require(!useDisk, "Off-heap storage level does not support using disk")
-    require(!useMemory, "Off-heap storage level does not support using heap memory")
     require(!deserialized, "Off-heap storage level does not support deserialized storage")
```

Comment:
Unfortunately, it's not going to be super easy to do this right now because of how StorageLevel serves the dual purpose of being both a specification of how blocks should be stored and a description of how they're actually stored (i.e. an "effective storage level"). If we added this constraint, the logic for constructing effective storage levels would become much more complicated.

Therefore, I think we should address this issue later and leave the code as-is for now.